sql/gen-sql-markdown.py

```diff
@@ -47,6 +53,18 @@ def _list_function_infos(jvm):
     return sorted(infos, key=lambda i: i.name)
 
 
+def _list_sql_configs(jvm):
+    sql_configs = [
+        SQLConfEntry(
+            name=_sql_config._1(),
+            default=_sql_config._2(),
```

Comment:
I noticed some strange values here like `67108864b`, where it looks like we need to do some kind of type casting somewhere, perhaps on the Scala side.

Same goes for stuff like `<undefined>` and `<value of ...>`, which I am handling down in `generate_sql_configs_table()` using regular expressions as a formatting step.

Reply:
I vaguely suspect it's because of some default values like `Option`. I will take a close look within few days.

Reply:
Okay `67108864b` isn't actually particularly wrong. It just converted bytes configuration to string byte representation. I think we should the default value to strings so that it doesn't get into byte string representation. For example,

```diff
--- a/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
+++ b/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala
@@ -1013,7 +1013,7 @@ object SQLConf {
       "This configuration is effective only when using file-based sources such as Parquet, JSON " +
       "and ORC.")
     .bytesConf(ByteUnit.BYTE)
-    .createWithDefault(128 * 1024 * 1024) // parquet.block.size
+    .createWithDefaultString("128MB") // parquet.block.size
```

Reply:
`<undefined>` is about optional configurations. So, it's only effective when it's set. I think we can just convert such value to things like `N/A`, no default value or anything most commonly used terms in such documentation.

Reply:
`<value of ...>` seems the case of fallback configuration where the default value is from another configuration. I think it might be better to explicitly mention this.

Reply:
So would you want me to adjust the various defaults to use `.createWithDefaultString()` as part of this PR? And if so, should I also fix minor formatting issues in the docstrings as I come across them (e.g. converting `'spark.config.name'` to `` `spark.config.name` ``)?

As for `<undefined>` and `<value of ...>`, I am parsing them and translating them into plain strings that won't muck up the HTML. When you say 

> I think it might be better to explicitly mention this.

what do you mean specifically?

Reply:
> So would you want me to adjust the various defaults to use `.createWithDefaultString()` as part of this PR? And if so, should I also fix minor formatting issues in the docstrings as I come across them (e.g. converting `'spark.config.name'` to `` `spark.config.name` ``)?

Are there a lot of instances of `'spark.config.name'` to `` `spark.config.name` ``? If there are too many, let's do it in a separate PR.

For `.createWithDefaultString()`, yes, there seems only 5 ~ 6 instances (except `.internal()` configurations). Looks fairly easy to change.

> what do you mean specifically?

I meant, we might have to explicitly say, this configuration is optional or default is unknown, for instance, `N/A`, an empty string, or `No default value`, instead of `<undefined> `.

And, ``Defaults to `spark.blah.blah` `` instead of `value of ...` as an example.

Reply:
> So would you want me to adjust the various defaults to use `.createWithDefaultString()` as part of this PR? And if so, should I also fix minor formatting issues in the docstrings as I come across them (e.g. converting `'spark.config.name'` to `` `spark.config.name` ``)?

Are there a lot of instances of `'spark.config.name'` to `` `spark.config.name` ``? If there are too many, let's do it in a separate PR.

For `.createWithDefaultString()`, yes, there seems only 5 ~ 6 instances (except `.internal()` configurations). Looks fairly easy to change.

Reply:
> what do you mean specifically?

Ah, okay. I was thinking of a better expressions for `<undefined>` and `<value of ...>` such as N/A, No default value, etc.

I just checked the other configurations in https://spark.apache.org/docs/latest/configuration.html. Seems it's `(none)` for `<undefined>`  and  `(value of ...)` for `<value of ...>`.

Reply:
> I just checked the other configurations in https://spark.apache.org/docs/latest/configuration.html. Seems it's `(none)` for `<undefined>` and `(value of ...)` for `<value of ...>`.

There is some inconsistency there, as we also have `None` and `NONE`, but I'll go with your suggestions as they are the consistent with the majority of configuration lines.